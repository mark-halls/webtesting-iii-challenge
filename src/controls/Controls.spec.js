// Test away!
import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";

import Controls from "./Controls";

afterEach(cleanup);

test("it renders the controls", () => {
  const { getByText } = render(<Controls closed={false} locked={false} />);

  expect(getByText(/lock gate/i)).toMatchSnapshot();
  expect(getByText(/close gate/i)).toMatchSnapshot();
});

test("it locks the gate", () => {
  const toggleLockedMock = jest.fn();
  const { getByText } = render(
    <Controls closed={true} locked={false} toggleLocked={toggleLockedMock} />
  );
  fireEvent.click(getByText(/lock gate/i));
  expect(toggleLockedMock).toHaveBeenCalled();
});

test("it closes the gate", () => {
  const toggleClosedMock = jest.fn();
  const { getByText } = render(
    <Controls closed={false} locked={false} toggleClosed={toggleClosedMock} />
  );

  fireEvent.click(getByText(/close gate/i));
  expect(toggleClosedMock).toHaveBeenCalled();
});

test("it can't open the gate if locked", () => {
  //check if open button disabled
  const { getByText } = render(<Controls closed={true} locked={true} />);

  expect(getByText(/open gate/i).disabled === true);
});

test("it can't lock the gate if open", () => {
  //check if lock button disabled
  const { getByText } = render(<Controls closed={false} locked={false} />);

  expect(getByText(/lock gate/i).disabled === true);
});
