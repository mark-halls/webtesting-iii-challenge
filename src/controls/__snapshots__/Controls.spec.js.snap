// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the controls 1`] = `
<button
  class="toggle-btn"
  disabled=""
>
  Lock Gate
</button>
`;

exports[`it renders the controls 2`] = `
<button
  class="toggle-btn"
>
  Close Gate
</button>
`;
